packages/iov-ethereum/src/serialization.ts

```diff
@@ -184,6 +239,70 @@ export class Serialization {
           s,
         );
       }
+    } else if (isSwapOfferTransaction(unsigned)) {
+      if (!unsigned.fee || !unsigned.fee.gasPrice) {
+        throw new Error("fee.gasPrice must be set");
+      }
+      const gasPriceHex = encodeQuantityString(unsigned.fee.gasPrice.quantity);
+      if (!unsigned.fee.gasLimit) {
+        throw new Error("fee.gasLimit must be set");
+      }
+      const gasLimitHex = encodeQuantityString(unsigned.fee.gasLimit.quantity);
+
+      if (!unsigned.swapId) {
+        throw new Error("No swap ID provided");
+      }
+
+      if (!unsigned.contractAddress || !isValidAddress(unsigned.contractAddress)) {
+        throw new Error("Invalid contract address");
+      }
+
+      if (!isValidAddress(unsigned.recipient)) {
+        throw new Error("Invalid recipient address");
+      }
+
+      if (unsigned.memo) {
+        throw new Error("Memo cannot be serialized in an atomic swap offer transaction");
+      }
+
+      if (!(unsigned.timeout as BlockHeightTimeout).height) {
+        throw new Error("Timeout must be specified as a block height");
+      }
+
+      if (unsigned.amounts.length !== 1 || unsigned.amounts[0].tokenTicker !== constants.primaryTokenTicker) {
+        // TODO: Remove when ERC20 swaps are supported
+        throw new Error("Only ETH atomic swap offers are currently supported");
+      }
```

Comment:
`unsigned.amounts.length !== 1` should have its own check since we have no plan to support multiple amounts in Ethereum any time soon, so it should stay.